File: app/src/main/java/com/example/socialapp/readapters/BleepsAdapter.java
package com.example.socialapp.readapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.socialapp.R;
import com.example.socialapp.models.Bleep;
import com.mikhaellopez.circularimageview.CircularImageView;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;
import java.util.Locale;

public class BleepsAdapter extends RecyclerView.Adapter<BleepsAdapter.BleepViewHolder> {

    private final List<Bleep> bleepList;
    private final Context context;
    private final OnBleepClickListener bleepClickListener;

    public BleepsAdapter(List<Bleep> bleepList, Context context, OnBleepClickListener bleepClickListener) {
        this.bleepList = bleepList;
        this.context = context;
        this.bleepClickListener = bleepClickListener;
    }

    @NonNull
    @Override
    public BleepViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.bleep_row, parent, false);
        return new BleepViewHolder(v, bleepClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull BleepViewHolder holder, int position) {
        Bleep bleep = bleepList.get(position);

        Glide.with(context).load(bleep.getUser().getImage()).into(holder.bleepPicView);
        holder.bleepNickView.setText(bleep.getUser().getNick());
        holder.bleepTimeView.setText(Bleep.timeStringFromMillis(bleep.getTimeMillis()));
        holder.bleepContentView.setText(bleep.getContent());
    }

    @Override
    public int getItemCount() {
        return bleepList.size();
    }

    public static class BleepViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        private final CircularImageView bleepPicView;
        private final TextView bleepNickView;
        private final TextView bleepTimeView;
        private final TextView bleepContentView;
        private final OnBleepClickListener bleepClickListener;

        public BleepViewHolder(@NonNull View itemView, OnBleepClickListener bleepClickListener) {
            super(itemView);
            this.bleepClickListener = bleepClickListener;
            bleepPicView = itemView.findViewById(R.id.bleepDetailsPic);
            bleepNickView = itemView.findViewById(R.id.bleepNickTextView);
            bleepTimeView = itemView.findViewById(R.id.bleepTimeTextView);
            bleepContentView = itemView.findViewById(R.id.bleepContentTextView);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            bleepClickListener.onBleepClick(getAdapterPosition());
        }
    }

    public interface OnBleepClickListener {
        void onBleepClick(int position);
    }
}

File: app/src/main/java/com/example/socialapp/readapters/BleepsAdapter.kt
package com.example.socialapp.readapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.socialapp.R
import com.example.socialapp.models.Bleep
import com.example.socialapp.readapters.BleepsAdapter.BleepViewHolder
import com.mikhaellopez.circularimageview.CircularImageView

class BleepsAdapter(private val bleepList: List<Bleep>, private val context: Context, private val bleepClickListener: OnBleepClickListener) : RecyclerView.Adapter<BleepViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BleepViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.bleep_row, parent, false)
        return BleepViewHolder(v, bleepClickListener)
    }

    override fun onBindViewHolder(holder: BleepViewHolder, position: Int) {
        val bleep = bleepList[position]
        Glide.with(context).load(bleep.user?.image).into(holder.bleepPicView)
        holder.bleepNickView.text = bleep.user?.nick
        holder.bleepTimeView.text = Bleep.timeStringFromMillis(bleep.timeMillis)
        holder.bleepContentView.text = bleep.content
    }

    override fun getItemCount(): Int {
        return bleepList.size
    }

    class BleepViewHolder(itemView: View, private val bleepClickListener: OnBleepClickListener) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        val bleepPicView: CircularImageView = itemView.findViewById(R.id.bleepDetailsPic)
        val bleepNickView: TextView = itemView.findViewById(R.id.bleepNickTextView)
        val bleepTimeView: TextView = itemView.findViewById(R.id.bleepTimeTextView)
        val bleepContentView: TextView = itemView.findViewById(R.id.bleepContentTextView)

        override fun onClick(v: View) {
            bleepClickListener.onBleepClick(adapterPosition)
        }

        init {
            itemView.setOnClickListener(this)
        }
    }

    interface OnBleepClickListener {
        fun onBleepClick(position: Int)
    }
}